topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorBaseIT.java

```diff
@@ -226,10 +238,21 @@ protected void startTopicOperator(StrimziKafkaCluster kafkaCluster) throws Inter
         LOGGER.info("Started Topic Operator");
     }
 
-    protected Map<String, String> topicOperatorConfig(StrimziKafkaCluster kafkaCluster) {
-        Map<String, String> m = new HashMap<>();
-        m.put(Config.KAFKA_BOOTSTRAP_SERVERS.key, kafkaCluster.getBootstrapServers());
-        m.put(Config.ZOOKEEPER_CONNECT.key, kafkaCluster.getZookeeper().getHost() + ":" + kafkaCluster.getZookeeper().getFirstMappedPort());
+    protected <T extends Startable> Map<String, String> topicOperatorConfig(final T strimziContainer) {
```

Comment:
Do we really need the type parameter? Can't the value parameter have type `Startable`?

It any case it feels a bit rubbish to require a dependency on `testcontainers`. We don't actually need any of the methods of `Startable`, we're just using it because it's a convenient supertype of both `StrimziKafkaCluster` and `StrimziKafkaContainer`. Do you think it's worth having a common superinterface for `StrimziKafkaCluster` and `StrimziKafkaContainer` in `strimzi-test-container`, to allow precisely this kind of abstraction in users of `strimzi-test-container`? It would also be more typesafe, since trying to call this method with any old `Startable` would then be a compile time error.

Reply:
And looking at how you're using `strimziContainer` it seems this interface could just expose a `getBootstrapServers` and `getZookeeperConnectString()` (and maybe some methods for the kraft case?) thus avoiding the `instanceof` entirely.

Reply:
Yes, I think this is a good idea to create an interface for such containers, but it would mean another release to get this PR merged. But overall I think it could wait until next release of test container :)